cmd/promtool/resource.go

```diff
@@ -30,7 +30,7 @@ type resourceSaverConfig struct {
 }
 
 type resourceSaver struct {
-	writer            archiveWriter
+	archiver
 	httpClient        httpClient
 	fileNameToRequest map[string]*http.Request
```

Comment:
wouldn't this be better reversed?
`requestToFile map[*http.Request]string`
we are making a request and writing to a file right?

Reply:
This makes more sence!